src/components/10-atoms/heading/index.js

```diff
@@ -0,0 +1,90 @@
+import { LitElement, html, css, unsafeCSS } from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+
+/* eslint-disable import/no-extraneous-dependencies */
+import defineOnce from '../../../utils/define-once';
+import { applyDefaults } from '../../../utils/with-react';
+import styles from './index.scss';
+
+class AXAHeading extends LitElement {
+  static get tagName() {
+    return 'axa-heading';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    // Define properties and types
+    return {
+      variant: { type: String },
+      rank: { type: Number },
+    };
+  }
+
+  constructor() {
+    super();
+    // this functions applies default values per type and verifies if
+    // the HTML attribute has been set before defining the custom element
+    applyDefaults(this);
+  }
+
+  render() {
+    const classes = classMap({
+      'a-heading--secondary': this.variant === 'secondary',
+    });
+
+    switch (this.rank) {
+      case 1:
```

Comment:
I would use a loop here. So that you don't have to write 6x the same stuff

Reply:
This is impossible as I see it. Or do you have an idea how to approach that?

Reply:
No not really, I think there would be the same problem as with the approach of Dany.

Reply:
@raphaellueckl @NoahLiechti this should work. Agree, remove the switch case please

```
<h${this.rank} class="a-heading ${classes}">
  <slot></slot>
</h${this.rank}>
```

Reply:
There must be a way.

Reply:
Or somthing like

```
const opening = `<${rank} class="a-heading ${classes}">`;
const closing = `</${rank}>`;

html`${opening}<slot></slot>${closing}`
```

?

Reply:
According to this question, there is not: https://stackoverflow.com/questions/59259400/is-there-a-way-to-make-tags-in-lit-html-templates-dynamic

but maybe we can do a workaround

Reply:
We found a workaround.